descqa/SizeDistribution.py

```diff
@@ -0,0 +1,124 @@
+        else:
+            self.validation_data = self.get_validation_data(observation)
+
+        self._color_iterator = ('C{}'.format(i) for i in count())
+        self._other_kwargs = kwargs        
+
+    def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
+        # update color and marker to preserve catalog colors and markers across tests
+        catalog_color = next(self._color_iterator)
+
+        # check catalog data for required quantities
+        key = catalog_instance.first_available(*self.acceptable_keys)
+        if not key:
+            summary = 'Missing required quantity' + ' or '.join(['{}']*len(self.acceptable_keys))
+            return TestResult(skipped=True, summary=summary.format(*self.acceptable_keys))
+
+        # get data
+        catalog_data = catalog_instance.get_quantities(key)
+        sizes = catalog_data[key]
+        good_data_mask = np.logical_not(np.logical_or(np.isinf(sizes), np.isnan(sizes)))
+        sizes = sizes[good_data_mask]
+        non_neg_mask = sizes>0
+        if np.sum(non_neg_mask)>0:
+            print('Warning: some sizes were negative or zero; these are being masked')
+            sizes = sizes[non_neg_mask]
+        min_sizes = np.min(sizes)
+        max_sizes = np.max(sizes)
+        
+        # Compute N(size) and its slope at the small end.
+        # Things seem to be roughly linear where N(size)>0.5*Ntot so use those points.
+        # Get ~20 points for the line fit, but compute the whole graph
+        median = np.median(sizes)
+        n_bins = int(20*(max_sizes-min_sizes)/(median-min_sizes)) 
+        N, bin_edges = np.histogram(sizes, bins=n_bins)
+        sumM = np.histogram(sizes, weights=sizes, bins=bin_edges)[0]
+        sumM2 = np.histogram(sizes, weights=sizes**2, bins=bin_edges)[0]
+        size_pts = get_opt_binpoints(N, sumM, sumM2, bin_edges)
+        diff = size_pts[1:] - size_pts[:-1]
+        if not np.all(diff>=0):
+            # Sparsely populated bins sometimes cause problems for
+            # get_opt_binpoints; replace with the dumb solution
+            size_pts = 0.5*(bin_edges[:-1]+bin_edges[1:])
+
+        mask = size_pts<median
+
+        # Normalize so we can compare datasets of different sizes
+        cumul_N_norm = np.array(
+                [1.0*np.sum(N[-i-1:]) for i in range(len(N))], dtype=float
+            )[::-1]/np.sum(N)
+        data_slope, data_intercept = np.polyfit(size_pts[mask], cumul_N_norm[mask], 1)
+        
+        # Compute the slope for the validation dataset in this size range.
+        # Copy the validation dataset so we can play with it
+        validation_data = self.validation_data.copy()
+        validation_mask = (validation_data[:,0] > min_sizes) & (validation_data[:,0]<median)
+        validation_data[:,1] /= validation_data[validation_mask,1][0]
+        validation_slope, validation_intercept = np.polyfit(validation_data[validation_mask, 0],
```

Comment:
`validation_intercept` is not used. You can change it to `_` or add `#pylint: disable=unused-variable` to this line to pass pylint check.